### ISSUE TITLE
Columns createdBy, createdDate, lastModifiedBy and lastModifiedDate should be filled automatically

### ISSUE THREAD

**Issue by glutengo** [jhipster/generator-jhipster-nodejs#236]
**Is your feature request related to a problem? Please describe.**
The base entity has the mentioned columns. However, these columns are never used when a new entity is written to the database, so they always have `null` as value.

**Describe the solution you'd like**
These columns should be filled automatically when an entity is saved. For the date columns, this could be achived by replacing the `Column()` decorator with `CreateDateColumn()` and `UpdateDateColumn()`(as describe [here](https://github.com/typeorm/typeorm/blob/master/docs/entities.md#special-columns)). For the user columns this could be handled somewhere in the entity service.

**Comment by glutengo** [comment]
Quick status update on this:

* Storing the created / last modified date was pretty simple and worked as expected.
* For storing the user, my initial approach was to inject the current request into the relevant service (namely UserService and EntityService), read the current user from the request (this has been set previously by the Passport JwtStrategy I think). But there are some issues:
    * If I do that, the affected services are re-generated for every request. This could have some impacts on performance (see [here])(https://guxi.me/posts/why-you-should-avoid-using-request-scope-injection-in-nest-js/))
    * In addition to that, this breaks the passport authentication strategy, as documented [here](https://github.com/nestjs/nest/issues/1870). I was able to work around this by using [request-scoped authentication strategies](https://docs.nestjs.com/security/authentication#request-scoped-strategies). I did make it past the e2e server tests and it worked for a locally generated application on my machine, but the e2e client tests with production builds are still broken with the error message from the mentioned issue.

I see three options to continue now:
* Further investigate why the client e2e tests are failing and find a solution. However the used setup might have performance impacts, so I am not sure if this is the right thing to do
* Do no try to inject the request in the services but pass the request (or the user) as a method parameter to the required methods instead. This is probably the safest way to go, but it will bloat the method signatures of many methods of the mentioned services
* Try to use a different mechanism like [zone.js](https://www.npmjs.com/package/zone.js?activeTab=readme), cls_hooked[https://www.npmjs.com/package/cls-hooked] or async_hooks[https://nodejs.org/api/async_hooks.html] as described [here](https://github.com/nestjs/nest/issues/173). This seems quite interesting but would mean that a new library is added (in case of zone.js) or we use an experimental API feature which may be subject to change in the future.

@amanganiello90 what do you think? Any suggestions?

**Comment by amanganiello90** [comment]
HI @glutengo you are rock!! Great work!!

However I don't understand your goal. Have you implemented the createdBy lastModifyed and so on filled automatically? That is the issue.

Now why you want to manipulate user and store it in the request?

**Comment by glutengo** [comment]
Hi @amanganiello90, thanks for getting back!

I have been able to implement createdDate and lastModifiedDate. I have not managed to implement createdBy and lastModifiedBy yet because of the problems I described above.

**Comment by amanganiello90** [comment]
I really appreciate your dedication to work in these days. So @mraible could we add a bug bounty on this issue (or on #214) in order to give a price for @glutengo collaboration? Thanks a lot

**Comment by glutengo** [comment]
@amanganiello90 thanks for your appreciation! I have found a solution now which works without request-scoped providers and without zone.js. I have adjusted the method signatures for the save and update methods of the services so that a creator / updater can be passed. I will open a PR for that now.

The NestJS lead developer has prepared a PR for integrating the async_hook API a while ago but refuses to merge it due to the experimental state of the API. I suggest to watch this and make use of it once it has been merged: https://github.com/nestjs/nest/pull/1407
